File: Python/Programmers/Leve1/[1차] 비밀지도.py
# def binary(num, n): # 이진수 변환
#     result = []
#     while num > 0:
#         result.append(num % 2)
#         num //= 2
    
#     result.extend([0] * (n-len(result))) # extend를 이용하여 리스트 마지막에 n의 개수만큼 0 추가
#     result.reverse()
#     return result

# def solution(n, arr1, arr2):
#     maps = []
#     for a, b in zip(arr1, arr2): # zip을 이용하여 묶음
#         maps.append(a|b) # or 비트연산으로 최종 지도를 구함
    
#     answer = []
#     for num in maps:
#         bina = binary(num, n) # 2진수 변환
#         tmp = ""
#         for b in bina:
#             if b == 1: tmp += "#" # 1이면 #, 0이면 공백
#             else: tmp += " "
        
#         answer.append(tmp)
#     return answer

def solution(n, arr1, arr2):
    answer = []
    for i, j in zip(arr1, arr2):
        map = str(bin(i|j)[2:]) # bin을 이용하여 2진수 변환 후 str을 통해 문자열로 변환
        map = map.rjust(n, "0") # 전체 자리 n을 확보한 후 map을 오른쪽 정렬 후 공백을 0으로 채움
        map = map.replace("1", "#") # 1을 #으로 교체
        map = map.replace("0", " ") # 0을 공백으로 교체
        answer.append(map)

    return answer

File: JavaScript/Programmers/KAKAO/[1차] 비밀지도.js
// 문제 요구에 따라 구현
// 지도 1 또는 지도 2 중 어느 하나라도 벽인 부분은 전체 지도에서도 벽, 모두 공백인 부분은 전체 지도에서도 공백
// 즉, 지도1, 2의 or 비트 연산을 통해 최종 지도를 구할 수 있다.

// function solution(n, arr1, arr2) {
//     return arr1.map((_, i) => { // map 메소드를 이용하여 최종지도를 한번에 구함
//         // arr1[i], arr2[i]의 or 비트연산 후 toString 메소드를 이용하여 2진수로 변환
//         // 이후 padStart 메소드를 이용하여 전체 길이가 n보다 작으면 앞에서 "0"을 채워 넣음
//         const result = (arr1[i] | arr2[i]).toString(2).padStart(n, "0");
//         // result 문자열을 전개문법을 이용하여 배열로 변환 후 각각의 요소가 "1"이면 "#", "0"이면 " "으로 매핑 후 다시 문자열로 변환
//         return [...result].map(bin => +bin ? "#" : " ").join("");
//     });
// }

function solution(n, arr1, arr2) {
    return arr1.map((_, i) =>
        // replace 메소드와 g(전역 플래그)를 이용하여 2진수로 변환된 문자열에서 0은 " "으로, 1은 "#"으로 변환
        (arr1[i] | arr2[i]).toString(2).padStart(n, "0").replace(/0/g, " ").replace(/1/g, "#")
   );
}
